packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,135 +5,119 @@
-        openOnFocus: true,
-        autoselect: false,
-        hint: false,
-        tabAutocomplete: false,
-      },
-      // Override algolia's default selection event, allowing us to do client-side
-      // navigation and avoiding a full page refresh.
-      handleSelected: (_input, _event, suggestion) => {
-        _event.stopPropagation();
-
-        // Use an anchor tag to parse the absolute url into a relative url
-        // Alternatively, we can use new URL(suggestion.url) but it's not supported in IE.
-        const a = document.createElement('a');
-        a.href = suggestion.url;
-
-        // Algolia use closest parent element id #__docusaurus when a h1 page title does
-        // not have an id, so we can safely remove it.
-        // See https://github.com/facebook/docusaurus/issues/1828 for more details.
-        const routePath =
-          `#__docusaurus` === a.hash
-            ? `${a.pathname}`
-            : `${a.pathname}${a.hash}`;
-        history.push(routePath);
-      },
-    });
-
-    if (focus) {
-      searchBarRef.current.focus();
-    }
-  }
-
-  const loadAlgolia = (focus = true) => {
-    if (algoliaLoaded) {
-      return;
-    }
-
-    Promise.all([import('docsearch.js'), import('./algolia.css')]).then(
-      ([{default: docsearch}]) => {
-        setAlgoliaLoaded(true);
-        window.docsearch = docsearch;
-        initAlgolia(focus);
-      },
-    );
-  };
+  const [isOpen, setIsOpen] = useState(false);
 
-  const toggleSearchInput = useCallback(() => {
-    loadAlgolia();
-
-    if (algoliaLoaded) {
-      searchBarRef.current.focus();
-    }
-
-    handleSearchBarToggle(!isSearchBarExpanded);
-  }, [isSearchBarExpanded]);
-
-  const handleSearchInputBlur = useCallback(() => {
-    handleSearchBarToggle(!isSearchBarExpanded);
-  }, [isSearchBarExpanded]);
+  const {
+    indexName,
+    appId = 'BH4D9OD16A',
+    apiKey,
+    searchParameters,
+  } = siteConfig.themeConfig.algolia;
+
+  const importDocSearchModalIfNeeded = useCallback(
+    function importDocSearchModalIfNeeded() {
+      if (DocSearchModal) {
+        return Promise.resolve();
+      }
+
+      return Promise.all([
+        import('@docsearch/react/modal'),
+        import('@docsearch/react/style'),
+      ]).then(([{DocSearchModal: Modal}]) => {
+        DocSearchModal = Modal;
+      });
+    },
+    [],
+  );
 
-  const handleSearchInput = useCallback((e) => {
-    const needFocus = e.type !== 'mouseover';
+  const onOpen = useCallback(
+    function onOpen() {
+      importDocSearchModalIfNeeded().then(() => {
+        setIsOpen(true);
+      });
+    },
+    [importDocSearchModalIfNeeded, setIsOpen],
+  );
 
-    loadAlgolia(needFocus);
-  });
+  const onClose = useCallback(
+    function onClose() {
+      setIsOpen(false);
+    },
+    [setIsOpen],
+  );
 
-  const handleSearchInputPressEnter = useCallback((e) => {
-    if (!e.defaultPrevented && e.key === 'Enter') {
-      navigateToSearchPage(e.target.value);
```

Comment:
It must be reverted.

Reply:
I see that a new search page was implemented in #2756 — we should think how to make these two features work together 🙂

If I refer to the original #2756 PR, here are the uses of this search page:

1. Show all results for a query
2. Support filters (for versioning)
3. Support Open Search

Those are features that we can either (1) support in DocSearch, or (2) bridge from DocSearch to the search page.

In DocSearch v3, the first hit is selected by default to allow to select the first result directly by hitting <kbd>Enter</kbd>. This is a good search UX practice for this kind of autocomplete experiences. Therefore, redirecting to the search page at this level does not make too much sense.

Instead, we could display a link (e.g., "Show all results") at the end of the DocSearch results that goes to the search page with the typed query.

Wdyt?

(Note that we've had discussions to support URL synchronization in DocSearch, which would cover this search page's main use case.)

Reply:
Addressed in c98d9a0b7dc91320e1cdccea36eafd39f43065a1.